### PR TITLE
feat: Update resource names to be more opr-paas specific

### LINKED ISSUE #147: feat: I find ClusterRole 'manager-role' and ClusterRoleBinding `manager-rolebinding` to be too Generic, add 'paas-'
# Feature Request

**Describe the Feature Request**

I find the ClusterRole 'manager-role' and ClusterRoleBinding `manager-rolebinding` to be too generic.

**Describe Preferred Solution**

I suggest to rename it to something like `paas-manager-role` and `paas-manager-rolebinding` to make it easier for people to find and to make it less generic to it has less chance to clash with any other operators on the kubernetes cluster.

**Related Code**

- manifests/rbac/role.yaml
- manifests/rbac/role_binding.yaml
- [row90]Makefile
```$(CONTROLLER_GEN) rbac:roleName=manager-role crd paths="./..." output:crd:artifacts:config=manifests/crds output:crd:artifacts:config=manifests/crds output:rbac:artifacts:config=manifests/rbac```

**If the feature request is approved, would you be willing to submit a PR?**
_(Help can be provided if you need assistance submitting a PR)_

- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by benoitschipper**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

Fixes: #147 

Feature PR for feature request 147

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by portly-halicore-76 — CHANGES_REQUESTED** [review]
Please update this role as well, this is not managed via kubebuilder, it's a one-off. https://github.com/belastingdienst/opr-paas/blob/main/manifests/rbac/leader_election_role_binding.yaml#L18

**Comment by benoitschipper** [comment]
Will do

**Comment by benoitschipper** [comment]
@portly-halicore-76 I checked and I can do that , do not see that this is being generated by the MakeFile. On the otherhand, I might then as well take the below with me:

- leader_election_role.yaml
- leader_election_role_binding.yaml
- service_account.yaml

and aas paas-* to them. And also re-update the ones I already committed.

Please let me know.

**Comment by portly-halicore-76** [comment]
> @portly-halicore-76 I checked and I can do that , do not see that this is being generated by the MakeFile. On the otherhand, I might then as well take the below with me:
> 
> * leader_election_role.yaml
> * leader_election_role_binding.yaml
> * service_account.yaml
> 
> and aas paas-* to them. And also re-update the ones I already committed.
> 
> Please let me know.

Please follow-up on this suggestion and specify the names in those yaml files as well.

**Comment by benoitschipper** [comment]
Will do

**Comment by benoitschipper** [comment]
Done

**Comment by benoitschipper** [comment]
I assume the tests fail as it tries to deploy it with a service account that does not exist now. Do we need to also change the manifests/manager/opr-paas-deployment.yaml to the correct SA?

**Comment by benoitschipper** [comment]
Suggest to change the e2e test, if we also want to change the deployment name: else the e2e test fails. What you say?

**Comment by portly-halicore-76** [comment]
> Suggest to change the e2e test, if we also want to change the deployment name: else the e2e test fails. What you say?

Great suggestion, please commit. You just have to update the deployments name in de e2e-tests-against-deployment in GA workflow. Than it should work.

Also please rebase on main to enable merging.

**Comment by benoitschipper** [comment]
Will do

**Comment by benoitschipper** [comment]
Should be good now

File: manifests/manager/opr-paas-deployment.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 kind: Deployment
 metadata:
-  name: controller-manager
+  name: paas-controller-manager
   namespace: paas-system
   labels:
-    control-plane: controller-manager
+    control-plane: paas-controller-manager
     app.kubernetes.io/name: deployment
     app.kubernetes.io/instance: controller-manager
```
Comment (portly-halicore-76) [inline manifests/manager/opr-paas-deployment.yaml]:
```suggestion
    app.kubernetes.io/instance: paas-controller-manager
```

**Review by portly-halicore-76 — APPROVED** [review]
Thanks you for your contribution.

**Pull request merged**